### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'tests')

### ISSUE THREAD

**Issue by MelaAxel** [boxblinkracer/cypress-testrail#61]
Seeing a new error today, after re-starting our implementation process:

Everything starts out normally:
```
Will leave all specs to run to filter at run-time
 Starting TestRail Integration v2.10.0
 ....................................................
 Cypress: 13.8.1
 Browser: unknown
 System: win32 (10.0.22631)
 Base URL: null
 TestRail Domain: {domain]
 TestRail Mode: Create Run
 TestRail Project ID: 153
 TestRail Milestone ID:
 TestRail Suite ID:
 TestRail Run Name: Regression PoC __datetime__
 TestRail Include All Cases: false
 Ignore pending tests: true
 Screenshots: true
 Include All Failed Screenshots: false
 TestRun created in TestRail: Regression PoC 5/3/2024, 12:39:03 PM
 New TestRail Run: R12694
GET /__/ 200 7.579 ms - -

```
All my tests run and pass so I close Cypress and get the following:

```
An error was thrown in your plugins file while executing the handler for the after:spec event.

The error we received was:

TypeError: Cannot read properties of undefined (reading 'tests')
    at Reporter._afterSpec ({path}\node_modules\cypress-testrail\src\Reporter.js:140:41)
    at Object.handler ({path}\node_modules\cypress-testrail\src\Reporter.js:80:24)
    at invoke ({path}\AppData\Local\Cypress\Cache\13.8.1\Cypress\resources\app\node_modules\@packages\server\lib\plugins\child\run_plugins.js:185:18)
    at {path}\AppData\Local\Cypress\Cache\13.8.1\Cypress\resources\app\node_modules\@packages\server\lib\plugins\util.js:59:14
    at tryCatcher ({path}\AppData\Local\Cypress\Cache\13.8.1\Cypress\resources\app\node_modules\bluebird\js\release\util.js:16:23)
    at Function.Promise.attempt.Promise.try ({path}\AppData\Local\Cypress\Cache\13.8.1\Cypress\resources\app\node_modules\bluebird\js\release\method.js:39:29)
    at Object.wrapChildPromise ({path}\AppData\Local\Cypress\Cache\13.8.1\Cypress\resources\app\node_modules\@packages\server\lib\plugins\util.js:58:23)
    at RunPlugins.execute ({path}\AppData\Local\Cypress\Cache\13.8.1\Cypress\resources\app\node_modules\@packages\server\lib\plugins\child\run_plugins.js:164:21)
    at EventEmitter.<anonymous> ({path}\AppData\Local\Cypress\Cache\13.8.1\Cypress\resources\app\node_modules\@packages\server\lib\plugins\child\run_plugins.js:56:12)
    at EventEmitter.emit (node:events:519:28)
    at EventEmitter.emit (node:domain:488:12)
    at process.<anonymous> {path}\AppData\Local\Cypress\Cache\13.8.1\Cypress\resources\app\node_modules\@packages\server\lib\plugins\util.js:33:22)
    at process.emit (node:events:519:28)
    at process.emit (node:domain:488:12)
    at process.emit.sharedData.processEmitHook.installedValue [as emit] ({path}\AppData\Local\Cypress\Cache\13.8.1\Cypress\resources\app\node_modules\@cspotcode\source-map-support\source-map-support.js:745:40)
    at emit (node:internal/child_process:951:14)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
Error:
      Expected DataContext to already have been set via setCtx. If this is a
      testing context, make sure you are calling "setCtx" in a before hook,
      otherwise check the application flow.

    at c (<embedded>:4315:138737)
    at Object.u (<embedded>:4359:75697)
    at Object.<anonymous> (<embedded>:4392:28989)
    at Object.tryCatcher ({path}\AppData\Local\Cypress\Cache\13.8.1\Cypress\resources\app\node_modules\bluebird\js\release\util.js:16:23)
    at Object.execute ({path}\AppData\Local\Cypress\Cache\13.8.1\Cypress\resources\app\node_modules\bluebird\js\release\method.js:15:34)
    at C.close (<embedded>:4450:26869)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Comment by MelaAxel** [comment]
Looks like its an issue on my side as I tried reverting to the 2.9 version and I still get the error - Trying to figure it out, since we haven't changed anything since this was working last (except updating all dependencies)

**Comment by MelaAxel** [comment]
After additional debugging, I still pretty sure this is on my side but I cannot figure out where or why... So I'm looking for additional help; please and thank you

**Comment by MelaAxel** [comment]
A little more info/context:

- I am only seeing this when I run cypress open; this doesn't log or show when I run the spec directly in headless
- the test results are not being sent to testrail in either case
- the run in testrail IS being created as intended
- in either method, I am left with a created_run.json file at root

**Comment by MelaAxel** [comment]
Based on my various tests, I'm going to see if @boxblinkracer can shed any light (headbashingagainstwall)

**Comment by MelaAxel** [comment]
@boxblinkracer just wanted to check-in and see if we can help provide any additional info on this

**Comment by MelaAxel** [comment]
TL;DR This looks like an issue caused by our spec formatting; I'd love to see an update to allow the formatting we have implemented so we can use this plug! :)


I think I may have found the issue - it looks like its the way we write out our POM and reusable functions. Our typical spec calls to our function models so our spec doesn't contain the `it` blocks and we don't use `context` at all (ie:
```
import { Homepage } from './functions/homepage';
import { Login } from './functions/login';

describe('C701950: Failed login with no credentials', () => {
  Homepage.visitHomepage();
  Login.validateSignInPage();
  Login.noCredentials();
});
```

This produces the error (in `open` mode but not in `run` mode) and shows nothing added to the run:
```
Updating TestRail run R13205. Adding test cases: 
TestRun updated in TestRail: 13205
```


So I did a bunch of other testing, got frustrated, set this aside until last week ...and then it dawned on me this morning to try reformatting a test to try one last thing... so I wrote out the full steps into a test file using the full `describe/context/it` structure and it worked...
```
  10 passing (30s)

 Updating TestRail run R13205. Adding test cases: 671910, 671911
 TestRun updated in TestRail: 13205
 TestRail >> Sending case results to run R13205: C671910,C671911
 Results sent to TestRail R13205 for: C671910,C671911
```

Since we don't use `context`, I stripped those and... still worked and I got the same output (with the updated runId ;))

so... I've also tried adding `context` and moving `it` blocks... but those potential workarounds don't seem to do anything to 'help'
